### ISSUE TITLE
Feature request: review mode for preanalyzed games

### ISSUE THREAD

**Issue by siasio** [featurecat/lizzie#578]
I find Lizzie the best tool for on-line AI game analysis because of neat and intuitive coloring of moves. Usually when I want a thorough analysis of my game I use [GoReviewPartner](https://github.com/pnprog/goreviewpartner) instead but its native board display is not nice and I view the file with another sgf editor. During training I also use a [database released by Facebook](https://ai.facebook.com/tools/elf-opengo) with pro games analyzed by ELF. In both cases it is annoying that I need to look at the comments section to learn if a move was a mistake or not. If moves were colored like in Lizzie I could focus just on a board.

I would love to see a review mode in Lizzie which would take a preanalyzed game and restore bot suggestions in a form of colored moves on a board. I can see that it is related to two points mentioned by @featurecat in #505 namely reloading games saved using Lizzie and winrate-color-coding. If these two things are done in 0.7 release, then I believe that adding support for GRP commentaries and Facebook commentaries will be not difficult.

For a reference [here](https://drive.google.com/file/d/1UhEaO7MBSQjM7bHVwxHWU-sQmd3aP-bm/view) is the 2nd Lee SeDol vs. AlphaGo game commented by LZ with GRP and [here](https://drive.google.com/file/d/1IVozpsfj1KbkSjEoQZn70_MXZ8I0rFJW/view) is a commentary by ELF.

Thank you,
Siasio

**Comment by zsalch** [comment]
It is now possible to save the status of the previous analysis into the sgf file and prioritize the previous analysis results on the next read.
Please check the following two config options and try to use them.
![image](https://user-images.githubusercontent.com/42595514/61850036-cb1e1180-aee5-11e9-9db1-148fd1a91b93.png)

**Comment by siasio** [comment]
As I use Windows, I waited for the official release and now I see how it works. That's what I had in mind, thank you!

However, it works for me only after I analyzed a game, saved it and started reviewing immediately. Somehow if I close Lizzie, open it again and try to open that preanalyzed file it seems to be broken. It shows me only the first move. Is it a bug or do I do something wrong?
![Lizzie_test](https://user-images.githubusercontent.com/39811817/66044080-0f024800-e521-11e9-9d8f-4837f0402935.png)

**Comment by featurecat** [comment]
That's really strange, you're not doing anything wrong. Could you please upload for us the pre-analyzed sgf so we can take a look?

**Comment by featurecat** [comment]
Here's what happens when I try loading the saved sgf again on my computer.
![scr](https://user-images.githubusercontent.com/10515890/66076089-06af0a80-e52b-11e9-84d1-7c38dd582ab2.png)

**Comment by siasio** [comment]
That's the [sgf file](https://drive.google.com/file/d/1HOMOmDwIG8VDV7BLKtC_MJfQVSMqQO1d/view?usp=sharing) I got.

**Comment by Mbr18** [comment]
The same for my Windows PC. See attached .zip file.
[SGF analyzed.zip](https://github.com/featurecat/lizzie/files/3683065/SGF.analyzed.zip)

**Comment by featurecat** [comment]
I found the problem - it's a bug between localizations. Your versions are using the format 0,42 and 0,53 to represent numbers, while mine would use 0.42 and 0.53. That's why I didn't catch it, I didn't test file saving on European localizations. Let me see if I can find the code responsible.

**Comment by featurecat** [comment]
Found it, I think: https://github.com/featurecat/lizzie/blob/7d6c10035b03fbaeafb6810dffcd65323c279c19/src/main/java/featurecat/lizzie/rules/SGFParser.java#L282

I might be able to fix it right now.

**Comment by featurecat** [comment]
Of course, this makes me wonder, how many similar localization issues might we have... That is so dumb, Java!

**Comment by featurecat** [comment]
I fixed it, I think. Please help me test it, replace your .jar file with this one. 
[fixed-localization-sgf.zip](https://github.com/featurecat/lizzie/files/3683138/fixed-localization-sgf.zip)

I will update the official release 0.7.1 later this week when I have time and we fix some other minor bugs.

**Comment by siasio** [comment]
Hm, somehow the .jar file doesn't open. It is only 612 kB compared to 4172 kB of the original file

**Comment by featurecat** [comment]
oh - sorry, I reset my pc recently, a setting must have been messed up. Let me investigate...

**Comment by featurecat** [comment]
[lizzie-fixed-sgf-localization.zip](https://github.com/featurecat/lizzie/files/3683181/lizzie-fixed-sgf-localization.zip)
Here is the working version. Sorry for the incorrect version earlier.

**Comment by siasio** [comment]
It works! Thanks a lot :)

**Comment by featurecat** [comment]
Btw this version I accidentally included a pull request that makes the move suggestions have very slightly weird shadows when they overlap. Hope it isn't an issue, we will release 0.7.1 later this week and that should be fixed.

**Comment by Mbr18** [comment]
Does not work yet for my Windows PC. 
See attachment.
[TestSGF.zip](https://github.com/featurecat/lizzie/files/3683280/TestSGF.zip)

**Comment by featurecat** [comment]
Alright. Does it work with siasio's lizzie-test.sgf? Please let me know. And please double, triple check that you did replace the jar I just sent. I will look for the bug.

**Comment by featurecat** [comment]
When I ran this, it worked just fine on the Shin Jinseo game. Can you please run Lizzie from the commandline or terminal, and please send me the output it generates when you try to load this file? Thanks.

**Comment by featurecat** [comment]
the command is: java -jar lizzie.jar

**Comment by Mbr18** [comment]
After trying lizzie-test.sgf  I now understand how this feature works.
The analyzed SGF can only be used with Lizzie!

There no were no variations available because I opened the analyzed SGF with Sabaki. 
I guess this works as intended for Lizzie (but it is too bad it is useless for Sabaki and Hactar Go).

**Comment by featurecat** [comment]
a-ha. Yes, I can't control (easily) how the analysis features are implemented in Sabaki. I'm sure it could be supported in the future if we asked nicely :) They would probably want to modify the exact way we're storing the information. Which is subject to change, by the way.

The way we currently store the analysis is that we store the information Leela Zero lz-analyze is giving us. But that makes file sizes really big, so in the future we may switch to a more size-efficient format.

Glad you got it working!
